http_client.go

```diff
@@ -318,6 +319,14 @@ func (c *HTTPClient) Send(data []byte) (response []byte, err error) {
 		}
 	}
 
+	if (c.config.CloseConnectionOnResponseStatus > 0) {
```

Comment:
I think we can make this behaviour default is status is 400, because it is unexpected situation for the server and client. Let's remove this config option at all.